.github/workflows/nightly-pr-checks.yaml

```diff
@@ -0,0 +1,212 @@
+        run: make lint-feature-files
+
+      - name: Lint conflicts in source files
+        if: success() || failure()
+        run: make lint-conflicts
+
+      - name: Lint shell scripts
+        if: success() || failure()
+        run: make lint-shell
+
+      - name: Check manifests up to date
+        if: success() || failure()
+        run: |
+          set -e
+
+          make manifests
+          make primaza go-generate
+          git update-index -q --refresh
+          if ! git diff-index --quiet HEAD -- config/ ; then
+            echo "the following manifests are not up-to-date"
+            git diff config
+            exit 1
+          fi
+
+  unit:
+    name: Unit Tests
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Checkout Git Repository
+        uses: actions/checkout@v3
+
+      - name: Unit Tests with Code Coverage
+        run: |
+          make test
+
+  features:
+    name: Acceptance Tests Features
+    runs-on: ubuntu-20.04
+    outputs:
+      features: ${{ steps.features.outputs.features }}
+    steps:
+      - name: Checkout Git Repository
+        uses: actions/checkout@v3
+
+      - id: features
+        run: |
+          cd test/acceptance/features
+          FEATURES=$(ls *.feature | jq -R -s -c 'split("\n")[:-1]')
+          echo "features=$FEATURES" >> $GITHUB_OUTPUT
+
+  build:
+    name: Build images
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - make: primaza
+            image-name: primaza-controller
+          - make: agentapp
+            image-name: agentapp
+          - make: agentsvc
+            image-name: agentsvc
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Checkout Git Repository
+        uses: actions/checkout@v3
+
+      - name: Build images
+        run: |
+          make ${{ matrix.make }} docker-build IMG=${{ matrix.image-name }}:testing
+          docker save ${{ matrix.image-name }}:testing -o ${{ matrix.image-name }}.tar
+
+      - name: Upload image artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: images
+          path: ${{ matrix.image-name }}.tar
+
+  acceptance:
+    name: Acceptance Tests
+    runs-on: ubuntu-20.04
+    timeout-minutes: 90
+
+    needs: ["features", "build"]
+    strategy:
+      fail-fast: false
+      matrix:
+        feature: ${{ fromJSON(needs.features.outputs.features) }}
+
+    steps:
+      - name: Checkout Git Repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          architecture: "x64"
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
```

Comment:
it seems to me it is not needed. WDYT?

Reply:
We can probably take it out of a few places besides this as well, so let's give it a try.